src/main/java/com/google/devtools/build/lib/analysis/actions/FileWriteAction.java

```diff
@@ -70,6 +70,8 @@ public abstract class FileWriteAction extends AbstractFileWriteAction
   private static final int COMPRESS_CHARS_THRESHOLD = 256;
 
   private final boolean makeExecutable;
+  @Nullable
+  private final String mnemonic;
```

Comment:
nit: with strings I'd probably rather go with a default value, either "" or "FileWrite" than Nullable. That should be easier to handle.

PS AbstractFileWriteAction.getMnemonic already defaults to "FileWrite". So I believe this is the best choice.